api/state/option.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+import "github.com/ava-labs/hypersdk/vm"
+
+const Namespace = "corestate"
+
+func With() vm.Option {
+	return vm.NewOption(Namespace, func(v *vm.VM, _ []byte) error {
+		vm.WithVMAPIs(JSONRPCStateServerFactory{})(v)
```

Comment:
we should add a config that is enabled by default here and json unmarshals the supplied bytes to set the config

Reply:
I agree with your comment above https://github.com/ava-labs/hypersdk/pull/1447#issuecomment-2314126322 .

But I am not sure to understand this one. And not sure to understand the purpose of the config you talk about.
Regarding the code, enabling or disabling the apis is done in the controller `New()` factory (A file that will be written by the user). 
If the user does not call `state.With()` the api is disabled.

Reply:
Ah I see, sorry for the confusion. We're going in the direction of having the `vm/` package provide a default set of APIs. Right now, there's the controller file, which will construct a version with and without options, but ideally there's a simple `vm.DefaultNew` that will construct all of the APIs.

Could you enable the current state API on MorpheusVM and we'll add this to the default set, so that it works with the e2e tests?

To support VMs that do not use the same set of defaults, we'll need to break the test coverage down so that VMs can select the relevant set of tests.

Reply:
No problem !
If I am not wrong staye API is already enabled in the PR 👍